### ISSUE TITLE
Activity feed

### ISSUE THREAD

**Issue by MarieS-WiMLDS** [probabl-ai/skore#767]
### Is your feature request related to a problem? Please describe.

As a user, I want to see quickly items that are created, including warnings. 

### Describe the solution you'd like

This solution is very MVP. My goal is to create a spec that requires as little tech work, rework and object creation as possible at first.  

In a view called "Activity feed", I want to see all the items created & the warnings.  
The most recent should be on top.  
Whenever an item is updated, the older version is deleted from the feed so it appears on top only.  
If possible, the warnings shouldn't be items that appear in the items list on the left. 

#### UI mockups

Application must now have two (virtual) pages. One for the project view, one for the activity feed.
There URL must be:
- /#/project
- /#/activity

A new vertical left bar (looking like a workspace switcher) will act as the main app navigation.

![Flow _ Activity feed_light](https://github.com/user-attachments/assets/21079caa-37da-4a8a-97ae-0e56374f5c9a)
*light mode*
![Flow _ Activity feed_dark](https://github.com/user-attachments/assets/6d9360a5-81cc-4017-af80-ba9a7f000231)
*dark mode*

[Figma source](https://www.figma.com/design/Qzjyxv0lPDU2kKtsH9dcp3/Skore---Workstation---High-fidelity-mock-ups---Q3?node-id=4625-10896&t=TNudaJfZ46wHBIRa-1).

#### Project's activity as an HTTP API.

> Disclaimer: even if this a perfect use case for web sockets, we will stick to a REST endpoint.

##### Needs

We need to serialize a list of item version for the frontend to display them from most recent to oldest one.

> [!NOTE]  
> This means that: if a user asks for the last 5 minutes of activity, and one item has changed 2 times during this time frame (same identifier, two different values), they will receive 2 _versions_ of this item.

##### Proposed API

`GET` `/api/project/activity?after={creation_datetime}` (fetch the project activity)

###### Parameters

| name    | type     | data type                         | description                                                                |
|---------|----------|-----------------------------------|----------------------------------------------------------------------------|
| `after` | optional | datetime as UTC string (iso 8601) | A search filter (every item that have been added after the given datetime). Adding this parameter will allow frontend to only request new data after the initial call. |

###### Response

| http code | content-type                     | response              | description                                    |
|-----------|----------------------------------|-----------------------|------------------------------------------------|
| `200`     | `application/json;charset=UTF-8` | `[{...}, {...}, ...]` | A list of item ordered by descending `create_at` datetime. |

###### Data layout

```json
[
  {
    "name": "md_item",
    "media_type": "text/markdown",
    "value": "# hello",
    "updated_at": "2020-07-10 15:00:00.000"
  },
  {
    "name": "df_item",
    "media_type": "application/vnd.dataframe+json",
    "value": {},
    "updated_at": "2020-07-10 14:00:00.000"
  },
  //...
]
```

### Additional context

Here is a list of questions that are either on hold, or answered in this specs in order to go towards an MVP that we can demo and experience, but not necessarily with the best answer: 
1. Should this activity feed be in a view or in another object?
2. Should we enable the user to quickly send items to a given view from the feed?
3. What happens to items which are updated? They are already in the feed, and we allow only once an item to appear. For now, older versions disappear and the most recent appears at the top. 

### Ready State Checklist
- [x] Specification complete and validated
- [x] #768 -> implementation #848 
- [x] #814 -> implementation #857
- [x] Priority assigned
- [x] Team reviewed and approved
- [ ] Effort estimated (optional)

**Comment by tuscland** [comment]
> Should this activity feed be in a view or in another object?

To me, this is a dedicated space, ping @anasstarfaoui.

> Should we enable the user to quickly send items to a given view from the feed?

This is a feature for a future iteration.

> What happens to items which are updated? They are already in the feed, and we allow only once an item to appear. For now, older versions disappear and the most recent appears at the top.

You offer a first approach, and we can iterate on that.

**Comment by rouk1** [comment]
Ok for me, only caveat is that `train_test_split` does not have concrete item associated to id. Hence it will never goes to the UI.

**Comment by MarieS-WiMLDS** [comment]
What I liked about having the warning in activity feed is that it can be presented nicely, with a nice police, in a paragraph instead of a line, etc.  
However, it's true that I don't think flying warnings, without their context, are interesting for the user. Plus, when outside an item, we cannot log when they are solved.  
Let's remove them from the first iteration.

**Comment by anasstarfaoui** [comment]
Okay then here is the link to the first iteration of activity feed https://www.figma.com/design/Qzjyxv0lPDU2kKtsH9dcp3/Skore---Workstation---High-fidelity-mock-ups---Q3?node-id=4609-7164&t=4BuhqNELqxRDndV2-1
